gatekeeper/http/http.go

```diff
@@ -28,11 +31,35 @@ type Gatekeeper struct {
 }
 
 // NewGatekeeper returns a new Gatekeeper.
-func NewGatekeeper(org, team string, cfg *config.Config, api *api.Client) *Gatekeeper {
+func NewGatekeeper(org, team, certpath, keypath string, cfg *config.Config, api *api.Client) (*Gatekeeper, error) {
 	server := &http.Server{
 		Addr: cfg.GatekeeperAddress,
 	}
 
+	if certpath == "" || keypath == "" {
+		log.Printf("Starting Gatekeeper without SSL")
+	} else {
+		keypair, err := tlsKeypair(certpath, keypath)
+		if err != nil {
+			return nil, err
+		}
+
+		tlsConfig := &tls.Config{
+			MinVersion:               tls.VersionTLS12,
+			CurvePreferences:         []tls.CurveID{tls.CurveP521, tls.CurveP384, tls.CurveP256},
```

Comment:
I found this configuration based on some research online, but it may be better to remove `CurvePreferences` (maybe even `CipherSuites`, I think `PreferServerCipherSuites` can be removed safely in any case) from the config. Based on https://github.com/golang/go/blob/46f4bfb2d17a3ccb4b3207d086a90cac3c00ea2f/src/crypto/tls/common.go#L688 it looks like the defaults are `[]CurveID{X25519, CurveP256, CurveP384, CurveP521}` for `CurvePreferences`. Defaults for  `CipherSuites` depend on the hardware (https://github.com/golang/go/blob/46f4bfb2d17a3ccb4b3207d086a90cac3c00ea2f/src/crypto/tls/common.go#L917)